crdppf/views/legal_documents.py

```diff
@@ -107,111 +107,134 @@ def createNewDocEntry(request):
 
     return {'success':True}
 
-@view_config(route_name='getLegalDocuments', renderer='json')
-def getLegalDocuments(request):
-    """Gets all the legal documents related to a feature.
+@view_config(route_name='getLegalbases', renderer='json')
+def getLegalbases(params):
+    """Gets all the legal bases related to a feature.
+    Input: dict with params : topic, canton, municipalitynb, cadastrenb
+    Output: legalbases
     """
-    doclist = []
 
+    doclist = []
+    filterparams = {
+        'request': None,
+        'topic': None,
+        'layer': None,
+        'canton': None,
+        'muncipalitynb': None,
+        'cadastrenb': None
+    }
+            
     legalbases = {}
-    legalbases = DBSession.query(LegalBases).order_by(LegalBases.legalbaseid.asc()).all()
     
+    for param in params:
+        if params[param] is not None and param in filterparams.keys():
+            filterparams[param] = params[param]
+    
+    if param == 'topic':
+        legalbases = DBSession.query(LegalBases).filter_by(topic=params[param]).all()
+    else:
+        sdf
+        legalbases = DBSession.query(LegalBases).order_by(LegalBases.legalbaseid.asc()).all()
+
     for legalbase in legalbases :
-        doclist.append({
+        doclist.append({legalbase.legalbaseid:{
```

Comment:
Please begin the append content on a new line and add a space after each `:` in the append